slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -319,28 +329,47 @@ bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
 void SLAMServiceImpl::ProcessData() {
     // Set up and build the MapBuilder
     SetUpMapBuilder();
+    DetermineActionMode();
```

Comment:
[q] should DetemrineActionMode return the map filename if one is located? Thus removing need for second for loop below?

Reply:
I don't think it should return a map filename, since its job is merely to determine the action mode. I was also thinking about how to (or whether or not) to remove the second time we're looping over the map folder, but decided that I want to separate finding the action mode from finding the latest map filename from each other, and running the loop twice is not going to be a big overhead. So I chose modularity over performance improvement.

Reply:
got it, would be nice to not have to loop multiple times but due to the low number of expected maps in the folder and for the sake of code clarity lets keep it as is